backend/test/rpc-client.test.js

```diff
@@ -0,0 +1,43 @@
+import assert from 'node:assert'
+import { describe, it } from 'node:test'
+import { chainHeadTestData } from './test_data/chainHead.js'
+import { rawActorEventTestData } from './test_data/rawActorEvent.js'
+import { parse } from '@ipld/dag-json'
+import { getActorEvents, getActorEventsFilter, getChainHead } from '../lib/rpc-service/service.js'
+import { ClaimEvent } from '../lib/rpc-service/data-types.js'
+import { Value } from '@sinclair/typebox/value'
+
+describe('RpcApiClient', () => {
+  const makeRpcRequest = async (method, params) => {
+    switch (method) {
+      case 'Filecoin.ChainHead':
+        return parse(JSON.stringify(chainHeadTestData))
+      case 'Filecoin.GetActorEventsRaw':
+        return parse(JSON.stringify(rawActorEventTestData)).filter(e => e.height >= params[0].fromHeight && e.height <= params[0].toHeight)
+      default:
+        console.error('Unknown method')
+    }
+  }
+
+  it('test the retrieval of the chainHead', async () => {
+    const chainHead = await getChainHead(makeRpcRequest)
+    assert(chainHead)
+    const expected = parse(JSON.stringify(chainHeadTestData))
+    assert.deepStrictEqual(chainHead, expected)
+  })
+
+  it('test the retrieval of rawActorEvents', async () => {
+    Array.from({ length: 11 }, (_, i) => 4622129 + i).flatMap(async (blockHeight) => {
```

Comment:
Let's rewrite this using a for loop please.

To make the assertion errors easier to reason about, let's change this into a parameterized test, so that we have one test case for each block tested.

```js
for (let blockHeight = 4622129; blockHeight < 4622129 + 11; blockHeight++) {
   it(`retrieves rawActorEvents in block ${blockHeight}`, async () => {
     // ...
   })
}
```